cmd/tendermint/commands/rollback.go

```diff
@@ -0,0 +1,46 @@
+package commands
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/internal/state"
+)
+
+var RollbackStateCmd = &cobra.Command{
+	Use:   "rollback",
+	Short: "rollback tendermint state by one height",
+	Long: `
+Rollbacking state should be performed in the event of an incorrect applicate state transition
+whereby Tendermint has persisted an incorrect app hash and is thus unable to make 
+progress. This takes a state at height n and overwrites it with the state at height n - 1. 
+It is expected that the application also rolls back to height n - 1. No blocks are removed 
+so upon restarting Tendermint, the transactions in block n will be reexecuted against the 
+application.
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		height, hash, err := RollbackState(config)
+		if err != nil {
+			return err
```

Comment:
It looks like this returns errors all the way out of the internal packages that we maintain. A failure returned to a user with error messages might not be very clear. Can we add some additional information to this error as well as the errors inside of the `Rollback` function.

Reply:
Yeah I can wrap the errors to help trace them